UnrealPlugin/README.md

```diff
@@ -1,36 +1,112 @@
 
-First, when you are building dedicated server, make sure you add the following to DefaultGame.ini
+First, open your File Explorer and create a folder called “Plugins” in your games' root directory. Then drag the Playfab GSDK plugin file into the Plugins folder.
 
-```ini
-[/Script/UnrealEd.ProjectPackagingSettings]
-IncludeAppLocalPrerequisites=True
-```
+Right click on the .uproject file and choose the option to "Switch Unreal Engine version", which is how you can quickly check which Unreal Engine version you are currently using. 
+The popup seen below should appear. If you already see that the Unreal Engine version is source build, you don’t need to change anything, so click Cancel. If the Unreal version is not 
+currently the source build, select it from the dropdown list and then click OK. 
 
-If the category already exists in your DefaultGame.ini, then just add the second line to it. This ensures that all app local dependencies ship with the game as well.
+![image depicting a window that says "Select Unreal Engine Version"](Resources/UnrealTutorialImages/SelectUnrealEngineVersion.png)
+
+Right click on the .uproject file again and select “Generate Visual Studio Project Files".
 
-If you are using Continuous Integration (CI), then you could add it to your setup to only turn this flag on when building a dedicated server, so the additional dlls only get added if it is a dedicated server build.
+Then, double click to open the .uproject in Unreal.
 
-## Disabling plugins
-When you add Dedicated Server support to your project, you will definitely have created a \<projectname\>Server.Target.cs file.
+Using the Unreal development environment, go to Files->Create a new C++ class with whatever name you would like. In our example, we call this class MyGameInstance.
 
-In this file add the following lines to the constructor:
+Then close Unreal and generate project files in source build mode again.
+
+Then using Visual Studio, open those newly created files and add in the following methods for (MyGameInstance.cpp and MyGameInstance.h)
+
+#### MyGameInstance.cpp
 ```csharp
+// Fill out your copyright notice in the Description page of Project Settings.
+#include "MyGameInstance.h"
+#include "PlayfabGSDK.h"
+#include "ThirdPersonMP.h"
+
+void UMyGameInstance::Init()
+{
+	FPlayfabGSDKModule::Get().OnShutdown.BindUObject(this, &UMyGameInstance::OnGSDKShutdown);
+	FPlayfabGSDKModule::Get().OnHealthCheck.BindUObject(this, &UMyGameInstance::OnGSDKHealthCheck);
+}
+
+void UMyGameInstance::OnStart()
+{
+	UE_LOG(LogTemp, Warning, TEXT("Reached onStart!"));
+	if (!FPlayfabGSDKModule::Get().ReadyForPlayers())
+	{
+		FPlatformMisc::RequestExit(false);
+	}
+}
+
+void UMyGameInstance::OnGSDKShutdown()
+{
+	UE_LOG(LogTemp, Warning, TEXT("Shutdown!"));
+	FPlatformMisc::RequestExit(false);
+}
+
+bool UMyGameInstance::OnGSDKHealthCheck()
+{
+	UE_LOG(LogTemp, Warning, TEXT("Healthy!"));
+	return true;
+}
+```
+
+#### MyGameInstance.h:
+```csharp
```

Comment:
this is C++ not C#

Reply:
Yes, meant to fix this but I wasn't sure if GitHub would recognize c++ or cplusplus so I just used csharp to format it! I think I can use c_cpp so will change that.